routing/probability_estimator.go

```diff
@@ -0,0 +1,52 @@
+package routing
+
+import (
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// Estimator estimates the probability to reach a node.
+type Estimator interface {
+	// getPairProbability estimates the probability of successfully
+	// traversing to toNode based on historical payment outcomes for the
+	// from node. Those outcomes are passed in via the results parameter.
+	getPairProbability(now time.Time, results NodeResults,
+		toNode route.Vertex, amt lnwire.MilliSatoshi,
+		capacity btcutil.Amount) float64
+
+	// getLocalPairProbability estimates the probability of successfully
+	// traversing our own local channels to toNode.
+	getLocalPairProbability(now time.Time, results NodeResults,
+		toNode route.Vertex) float64
+
+	// config returns the current estimator's configuration.
+	config() EstimatorConfig
```

Comment:
Hmm, if we want to have a stricter or even entirely distinct config here, we can use type params to abstract away from this: 
```go
type Estimator[C any] interface {
     Config() T
}
```

Thought we might want to re-examine this method here itself: if we have an interface over the estimator itself, why do we need to be able to access the config? Given I'd assume that the actual estimator gets created elsewhere, then passed in to anyplace that accepts `Estimator`.

Reply:
Other methods that could make sense here would be something like `Purge`, `Import`, `Export`, etc. Though we don't necessarily need to fill them out in this PR.

Reply:
The issue is with `GetMissionControlConfig`, where we need to read out the config to be able to update it around the `lncli setmccfg/getmccfg` functionality. In intermediate stages of the PR I had different configs for the two estimators, but then reverted back because I thought it would be nice to be able to switch back and forth between different estimators without having to reconfigure parameters. So this way we store the entire config somewhere and can reuse it to switch, but still not convinced it's the best approach.